### ISSUE TITLE
Dynamically Generating a Pro-Form Using MOP Vars

### ISSUE THREAD

**Issue by brucewsinc** [websharks/s2member-kb#83]
##### KB Article Creation Checklist

- [x] Write initial draft for this KB Article; label this issue `draft` and either `questions` or `tutorials`
- [x] Add [required YAML configuration](https://github.com/websharks/team/wiki/KB-Articles-::-Required-YAML-Configuration)
- [x] Add Tags for this KB Article to the YAML config (see [YAML Keys (Explained)](https://github.com/websharks/team/wiki/KB-Articles-::-Required-YAML-Configuration#yaml-keys-explained))
- [x] Edit and finalize draft for publishing (remove `draft` label, add `draft-finalized` label)
- [x] Create Markdown file for this Article (remove `draft-finalized` label, add `pending` label)
- [x] Publish KB Article (remove `pending` label, add `published` label; leave this step for the Project Lead)

---

**Markdown File:** http://wsharks.com/1B8yVK8
**Published KB Article:** http://s2member.com/kb-article/dynamically-generating-a-pro-form-using-mop-vars/

---

**Comment by jaswsinc** [comment]
> Generating Pro Forms for Custom Capabilities

Wow! I don't think I've ever seen an article like this before. In fact, I don't think I've ever seen that technique used before; i.e., where the Pro Form is generated based on the MOP Vars. Cool!

I'd suggest that you work to enhance this article a bit. I'd start by listing the requirements (prerequisites) that must be satisfied before this article can become useful.

Also, I'd suggest that you move this from the `questions` category to the `tutorials` category. Re-title it "Dynamic Pro Form Generation, Based on MOP Vars". That sort of title will be scary enough; it will be an indication that you should only be reading this if you've completed a course on "how to be awesome".

---

Try to cleanup things like, "that will unlock locked down content when they are redirected from it.".

> locked down
should be locked-down

Having locked-down right after the word "unlock" makes it harder to read. I'd rephrase that.

**Comment by raamdev** [comment]
> should be locked-down

[Ha!](https://github.com/websharks/s2member-kb/issues/78#issuecomment-72789780) :-)

**Comment by raamdev** [comment]
@brucewsinc Great article and awesome idea for generating Pro-Forms based on MOP Vars!

**Comment by jaswsinc** [comment]
> when a User is redirected from it

I'd try to clarify what "it" is exactly in that sentence.

**Comment by jaswsinc** [comment]
@raamdev My feeling is that we should not publish this article, but what do you think? I'm a little concerned that this is going to generate more questions than it answers. It's a great article, but also very tricky.

**Comment by raamdev** [comment]
@jaswsinc We could just change the title to something like "Dynamically Generating a Pro-Form Using MOP Vars", which would be less inviting to questions and an overall feeling that _anything is possible!_

I'd also make it a tutorial instead of a question.

This article would be useful to have, IMO, as something to reference if somebody asks how it's done. I agree that it leaves the door wide open for all sorts of other questions, but we'll just draw the line and say (possibly even in a note at the bottom of the article) that we do not provide support for custom integrations or coding.

**Comment by jaswsinc** [comment]
@raamdev writes...
> We could just change the title to something like "Dynamically Generating a Pro-Form Using MOP Vars", which would be less inviting to questions and an overall feeling that anything is possible!

Copy that. Sounds good to me. Thanks.

**Comment by jaswsinc** [comment]
Cleaned this up a bit and published it as a tutorial. Thanks!

**Comment by Lifeischoice** [comment]
Good article...it will be helpful to add where to add the code...thanks

**Comment by raamdev** [comment]
@Lifeischoice Thank you for the suggestion. I've updated the KB Article.

**Comment by Lifeischoice** [comment]
Thanks...very useful.
